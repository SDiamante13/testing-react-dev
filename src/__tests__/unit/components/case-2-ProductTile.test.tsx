import React from 'react'
import { axe } from 'jest-axe'
import { render, screen } from '@testing-library/react'
import ProductTile from '../../../components/ProductTile'

describe('The <ProductTile /> component', () => {

  const defaultProductProps = {
    id: 1,
    image: 'shoes.png',
    name: 'Shoes',
    price: '42.00',
    brand: 'Nike',
    createdAt: '11/01/20',
    isActive: true,
    isNew: true,
    isSoldOut: false,
    priceUnformatted: 42.00
  }

  const setupProductTile = (props = defaultProductProps) => render(<ProductTile {...props}/>)

  it('should match snapshot', () => {
    const { asFragment } = setupProductTile();

    expect(asFragment).toMatchSnapshot();
  })

  it('renders a product tile with name, image and price', () => {
    setupProductTile();

    screen.getByText(defaultProductProps.name);
    screen.getByText(defaultProductProps.price);
    screen.getByAltText(defaultProductProps.name);
  })

  it('renders a product tile with name and price only', () => {
    setupProductTile({
      ...defaultProductProps,
      image: undefined
    });

    screen.getByText(defaultProductProps.name);
    screen.getByText(defaultProductProps.price);
    expect(screen.queryByTestId('ProductTileImage')).toBeNull();
  })

  it('has no accessibility violations', async () => {
    const {container} = setupProductTile();

    expect(await axe(container)).toHaveNoViolations();
  })
})
