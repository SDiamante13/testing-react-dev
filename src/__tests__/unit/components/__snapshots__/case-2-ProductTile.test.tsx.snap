// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ProductTile /> component renders a product tile with name, image and price 1`] = `[Function]`;

exports[`The <ProductTile /> component should match snapshot 1`] = `[Function]`;
